src/overlay021/pokedex_search.c

```diff
@@ -0,0 +1,1074 @@
+{
+    PokedexSearchSettings *searchSettings = Heap_AllocFromHeap(heapID, sizeof(PokedexSearchSettings));
+
+    GF_ASSERT(searchSettings);
+    memset(searchSettings, 0, sizeof(PokedexSearchSettings));
+
+    searchSettings->unk_00 = ov21_021D138C(param1);
+    searchSettings->unk_04 = ov21_021D13EC(param1);
+    UnkStruct_ov21_021E68F4 *v1 = ov21_021D1410(param1, 0);
+    searchSettings->unk_08 = v1->unk_00;
+
+    return searchSettings;
+}
+
+static UnkStruct_ov21_021D879C *ov21_021D8724(enum HeapId heapID, UnkStruct_ov21_021D0F60 *param1)
+{
+    UnkStruct_ov21_021D879C *v0 = Heap_AllocFromHeap(heapID, sizeof(UnkStruct_ov21_021D879C));
+
+    GF_ASSERT(v0);
+    memset(v0, 0, sizeof(UnkStruct_ov21_021D879C));
+
+    v0->unk_00 = ov21_021D13FC(param1);
+
+    return v0;
+}
+
+static UnkStruct_ov21_021D4660 *ov21_021D874C(enum HeapId heapID, UnkStruct_ov21_021D0F60 *param1)
+{
+    int v1 = ov21_021D87C8();
+
+    UnkStruct_ov21_021D4660 *v0 = Heap_AllocFromHeap(heapID, sizeof(UnkStruct_ov21_021D4660) * v1);
+
+    GF_ASSERT(v0);
+    memset(v0, 0, sizeof(UnkStruct_ov21_021D4660) * v1);
+
+    ov21_021D47F0(heapID, &v0[0], param1, (0x1 << 1));
+
+    return v0;
+}
+
+static void FreeSearchSettings(PokedexSearchSettings *searchSettings)
+{
+    GF_ASSERT(searchSettings);
+    Heap_FreeToHeap(searchSettings);
+}
+
+static void ov21_021D879C(UnkStruct_ov21_021D879C *param0)
+{
+    GF_ASSERT(param0);
+    Heap_FreeToHeap(param0);
+}
+
+static void ov21_021D87B0(UnkStruct_ov21_021D4660 *param0)
+{
+    GF_ASSERT(param0);
+
+    ov21_021D4660(&param0[0]);
+    Heap_FreeToHeap(param0);
+}
+
+static int ov21_021D87C8(void)
+{
+    return 1;
+}
+
+static int ResetSearchSettings(UnkStruct_ov21_021E6A68 *param0, void *param1)
+{
+    PokedexSearchSettings *searchSettings = param1;
+    int *v1 = Heap_AllocFromHeap(param0->heapID, sizeof(int));
+
+    GF_ASSERT(v1);
+    memset(v1, 0, sizeof(int));
+
+    param0->unk_08 = v1;
+
+    searchSettings->filterMethod = FM_ORDER;
+    searchSettings->sortOrder = SO_NUMERICAL;
+    searchSettings->filterName = FN_NONE;
+    searchSettings->typeFilter1 = FT_NONE;
+    searchSettings->typeFilter2 = FT_NONE;
+    searchSettings->filterForm = FF_NONE;
+    searchSettings->screenChange = FALSE;
+    searchSettings->applyFilter = FALSE;
+    searchSettings->screenState = SS_MENU;
+
+    return 1;
+}
+
+static int UpdateScreenState(UnkStruct_ov21_021E6A68 *param0, void *param1)
+{
+    PokedexSearchSettings *searchSettings = param1;
+
+    if (param0->unk_0C == 1) {
+        return 1;
+    }
+
+    if (param0->unk_10 == 1) {
+        return 0;
+    }
+
+    switch (param0->unk_00) {
+    case 0:
+
+        if (searchSettings->noneFound) {
+            searchSettings->noneFound = FALSE;
+        }
+
+        if (searchSettings->screenChange == TRUE) {
+            if (searchSettings->applyFilter == FALSE) {
+                *searchSettings->unk_00 |= (0x1 << 1);
```

Comment:
```suggestion
                *searchSettings->unk_00 |= (1 << 1);
```

That value also looks like a constant-candidate to me...